core/services/mavftp/main.py

```diff
@@ -0,0 +1,154 @@
+#!/usr/bin/env python
+
+import errno
+import os
+import time
+from argparse import ArgumentDefaultsHelpFormatter, ArgumentParser
+from typing import Any, Dict, Optional
+
+from fuse import FUSE, FuseOSError, LoggingMixIn, Operations
+from loguru import logger
+from pymavlink import mavutil
+
+from mavftp import FTPModule
+
+# this allows us to support acessing the special files
+base_file_paths = ["/", "@ROMFS", "@SYS"]
+
+
+class MAVFTP(LoggingMixIn, Operations):
+    def __init__(self, mav: Any) -> None:
+        self.mav = mav
+        self.files: Dict[str, Dict[str, int]] = {
+            path: {
+                "st_mode": 0o40755,
+                "st_nlink": 2,
+                "st_size": 0,
+                # we need some timestamp for filebrowser be happy
+                "st_ctime": int(time.time()),
+                "st_mtime": int(time.time()),
+                "st_atime": int(time.time()),
+            }
+            for path in base_file_paths
+        }
+        self.ftp = FTPModule(mav)
+
+    def fix_path(self, path: str) -> str:
+        if path.startswith("/@"):
+            return path[1:]
+        return path
+
+    def getattr(self, path: str, _fh: int = 0) -> Any:
+        path_fixed = self.fix_path(path)
+        logger.info(f"Fuse: getattr {path_fixed}")
+        if path_fixed in self.files:
+            return self.files[path_fixed]
+
+        parent_dir = ("/" + "/".join(path.split("/")[:-1])).replace("//", "/")
```

Comment:
can you add a comment about what is happening here ?

Reply:
is this literally `f"/{path}/../"` sanitized?

Reply:
oops, this is one of the things I meant to cleanup before opening it!